src/odemis/driver/tmcm.py

```diff
@@ -1725,40 +1778,58 @@ def _doMoveAbs(self, future, pos):
             end = 0 # expected end
             old_pos = self._applyInversion(self.position.value)
             moving_axes = set()
+            moving_do_axes = set()
             for an, v in pos.items():
-                aid = self._name_to_axis[an]
-                moving_axes.add(aid)
-                usteps = int(round(v / self._ustepsize[aid]))
-                # Actual position is the one used for absolute move, so need to
-                # always reset it when there is an encoder
-                self._resetEncoderDeviation(aid, always=True)
-                self.MoveAbsPos(aid, usteps)
-                # compute expected end
-                try:
-                    d = abs(v - old_pos[an])
-                    dur = driver.estimateMoveDuration(d, self.speed.value[an], self._accel[an])
-                except Exception: # Can happen if config is wrong and report speed or accel == 0
-                    logging.exception("Failed to estimate move duration")
-                    dur = 60
-                end = max(time.time() + dur, end)
-
-            self._waitEndMove(future, moving_axes, end)
+                # Check if it's a digital output
+                if an in self._name_to_do_axis:
+                    port = self._name_to_do_axis[an]
+                    if self._leds_on.value and v is not self._led_prot_do[port]:
```

Comment:
It should only check self._leds_on if `port` is in `self._led_prot_do`.
Also, if that's the case (and leds_on), then it shouldn't do anything (ever), expected updating the "latest requested" position. For niceness, you indeed can add an extra logging only in case the position is different from the current position.

Reply:
That's a bit tricky. I set the leds_on VA to True in requestRefSwitch before requesting the move to close the shutters, so if I don't have that extra condition, they are never closed.

The reason I do this is the following: Suppose the _leds_on VA is set right before the setio command for the reference leds and after we close the shutters. Now if we start an acquisition just after closing the shutters and before setting the va, the shutters are allowed to open again (because the va isn't set) and we will start referencing even though the shutters are open.

To mediate for this, we would need to use a lock every time we do referencing, which is maybe not a good idea.

Is there a better way of dealing with this? Or should I leave it as it was (with the second if condition)?